master/cache.lua

```diff
@@ -57,19 +85,27 @@ local function request(host, port, path)
 end
 
 
-local function fetch_and_cache_state_marathon()
+local function fetch_and_store_marathon_apps()
     -- Access Marathon through localhost.
     ngx.log(ngx.NOTICE, "Cache Marathon app state")
-    local appsRes, err = request("127.0.0.1", 8080, "/v2/apps?embed=apps.tasks&label=DCOS_SERVICE_NAME")
+    local appsRes, err = request("127.0.0.1", 8080,
+								 "/v2/apps?embed=apps.tasks&label=DCOS_SERVICE_NAME",
+                                 false)
 
     if err then
         ngx.log(ngx.NOTICE, "Marathon app request failed: " .. err)
+        if not cache_data("svcapps", nil) then
+            ngx.log(ngx.ERR, "Invalidating Marathon apps cache failed")
```

Comment:
What follows is only is an important point if we do this kind of cache invalidation at all: it seems that setting the cache value to `nil` without adjusting the corresponding "last refresh time" value seems inconsistent (but I might have missed something in the code).